### PR TITLE
Add honeybadger_events appender

### PR DESCRIPTION
**Pull request by trvsdnn**
### Changelog

Entry added:

> - Add appender for Honeybadger events/insights API

### Description of changes

Adds an additional Honeybadger appender `honeybadger_events` for their [events/insights API](https://docs.honeybadger.io/api/reporting-events/).

The appender uses the honeybadger gem and the `Honeybadger.event` method, which is intelligent about batching log lines and sending a single request.

### CODE REVIEW

**Comment by stympy** [comment]
I wonder if it would make sense to modify the existing appender instead of creating a new one. The non-exception branch of logic could use `Honeybadger.event` to send the payload.

**Comment by trvsdnn** [comment]
I considered that – it's a bit odd to use `semantic_logger` for exceptions when the honeybadger gem does it already. Unless I'm missing some clear benefit.

The problem IMO is that it's a change in behavior for anyone using it and would/should require a major release I would think.

This approach felt similar to the `NewRelic` vs `NewRelicLogs` appenders.

**Comment by stympy** [comment]
> I considered that – it's a bit odd to use `semantic_logger` for exceptions when the honeybadger gem does it already. Unless I'm missing some clear benefit.

Agreed :)

> The problem IMO is that it's a change in behavior for anyone using it and would/should require a major release I would think.
> 
> This approach felt similar to the `NewRelic` vs `NewRelicLogs` appenders.

Yeah, that makes sense.

File: lib/semantic_logger/appender/honeybadger_events.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        ::Honeybadger.event(event)
+
+        true
+      end
+
+      private
+
+      # Use Raw Formatter by default
+      def default_formatter
+        SemanticLogger::Formatters::Raw.new(time_key: :ts, time_format: :rfc_3339)
```
Comment (stympy) [inline lib/semantic_logger/appender/honeybadger_events.rb:57]:
```suggestion
        SemanticLogger::Formatters::Json.new(time_key: :ts, time_format: :rfc_3339)
```

Comment (trvsdnn) [reply lib/semantic_logger/appender/honeybadger_events.rb:57]: `Honeybadger.event` seems to use a string argument as the `event_type` and not treat it as an object.

```ruby
Honeybadger.event('{"foo":"bar"}')
```

results in

```
@ts "2024-06-12 19:34:05.000"
event_type "{"foo":"bar"}"
```

vs

```ruby
Honeybadger.event({foo:"bar"})
```

```
@ts "2024-06-12 19:37:29.000"
foo "bar"
```

Comment (stympy) [reply lib/semantic_logger/appender/honeybadger_events.rb:57]: Oh, right, ignore me :)

Comment (trvsdnn) [reply lib/semantic_logger/appender/honeybadger_events.rb:57]: Didn't even know it accepted a string... good to know!

**Comment by reidmorrison** [comment]
Understandably we are calling it Honeybadger Events, since it uses the Event API, but they refer to overall solution as Honeybadger Insights: https://docs.honeybadger.io/guides/insights/
Can we rename the appender to `HoneybadgerInsights` ?

For anyone that first comes across the appender, Honeybadger Insights would be simpler to understand.

File: lib/semantic_logger/appender/honeybadger_events.rb (hunk 1 above)
Comment (reidmorrison) [reply lib/semantic_logger/appender/honeybadger_events.rb:57]: Looking at the Honeybadger documentation for the event api, the first argument is the `event_type`, and the second argument is a `Hash`. https://docs.honeybadger.io/lib/ruby/getting-started/sending-events-to-insights/#manually-sending-events

We could make the `event_type` the value from `log.metric` when present and something like `event.semantic_logger` when a log entry has no metric. Following the examples from the documentation:  https://docs.honeybadger.io/lib/ruby/getting-started/sending-events-to-insights/#event-captures

**Comment by stympy** [comment]
> Understandably we are calling it Honeybadger Events, since it uses the Event API, but they refer to overall solution as Honeybadger Insights: https://docs.honeybadger.io/guides/insights/ Can we rename the appender to `HoneybadgerInsights` ?
> 
> For anyone that first comes across the appender, Honeybadger Insights would be simpler to understand.

👍

File: lib/semantic_logger/appender/honeybadger_events.rb (hunk 1 above)
Comment (stympy) [reply lib/semantic_logger/appender/honeybadger_events.rb:57]: It’s totally fine to pass a hash as the first argument and not specify an event type, if that makes more sense. 🙂

Comment (trvsdnn) [reply lib/semantic_logger/appender/honeybadger_events.rb]: It feels right to send `event_type` for instrumentation type events like `process_action.action_controller` – using the `honeybadger` gem to handle instrumentation gives you that.

For raw logging I don't like the idea of the appender deciding what you want to include in the log payload.

**Pull request merged**